go/roothash/api/api.go

```diff
@@ -170,55 +170,16 @@ type Genesis struct {
 	Blocks map[signature.MapKey]*block.Block `json:"blocks,omitempty"`
 }
 
-// ConsensusParameters are the roothash consensus parameters.
-type ConsensusParameters struct {
-	// RoundTimeout is the round timeout.
-	RoundTimeout time.Duration `json:"round_timeout"`
-
-	// TransactionScheduler is the transaction scheduler configuration.
-	TransactionScheduler TransactionSchedulerParameters `json:"txn_scheduler"`
-}
-
 // TransactionSchedulerAlgorithmBatching is the name of the batching algorithm.
 const TransactionSchedulerAlgorithmBatching = "batching"
 
-// TransactionSchedulerParameters is the transaction scheduler parameters.
-type TransactionSchedulerParameters struct {
-	// Algorithm is the transaction scheduling algorithm.
-	Algorithm string `json:"algorithm"`
-
-	// If using the "batching" algoritm, how long to wait for a scheduled batch.
-	BatchFlushTimeout time.Duration `json:"batch_flush_timeout"`
-
-	// If using the "batching" algorithm, what is the max size of a batch.
-	MaxBatchSize uint64 `json:"max_batch_size"`
-
-	// If using the "batching" algorithm, what is the max size of a batch
-	// in bytes.
-	MaxBatchSizeBytes uint64 `json:"max_batch_size_bytes"`
+// ConsensusParameters are the roothash consensus parameters.
+type ConsensusParameters struct {
 }
 
 // SanityCheck does basic sanity checking on the genesis state.
 func (g *Genesis) SanityCheck() error {
-	if g.Parameters.RoundTimeout < 1*time.Second {
-		return fmt.Errorf("roothash: sanity check failed: round timeout must be >= 1 second")
-	}
-
-	if g.Parameters.TransactionScheduler.Algorithm != TransactionSchedulerAlgorithmBatching {
-		return fmt.Errorf("roothash: sanity check failed: invalid txn sched algorithm")
-	}
-
-	if g.Parameters.TransactionScheduler.BatchFlushTimeout < 1*time.Second {
-		return fmt.Errorf("roothash: sanity check failed: batch flush timeout must be >= 1 second")
-	}
-
-	if g.Parameters.TransactionScheduler.MaxBatchSize < 1 {
-		return fmt.Errorf("roothash: sanity check failed: max batch size must be >= 1")
-	}
-
-	if g.Parameters.TransactionScheduler.MaxBatchSizeBytes < 1 {
-		return fmt.Errorf("roothash: sanity check failed: max batch size in bytes must be >= 1")
-	}
+	// TODO: Should we make per-runtime parameters sanity check here as well? -Matevz
```

Comment:
No, that needs to be done in the registry. Did you just remove the sanity checks instead of moving them?

Reply:
Yeah, I moved sanity checks away temporarily. I re-added them now to registry.